### ISSUE TITLE
Print layer output while training?

### ISSUE THREAD

**Issue by shyamupa** [fchollet/keras#1858]
Is there a way to print the output of a layer during training? I am trying to debug a attention model in which the loss is not at all changing during training. I have tried changing learning rates, checking data sanity, but to no success. I want to see whether the layer outputs actually make any sense during training (eg. is my Lambda layer slicing correctly etc.). 

I know I can print the weights by using,
```
Wr=m.nodes['Wr'].get_weights()
emb=m.nodes['emb'].get_weights()
```
but I want the output. When I do `m.nodes['emb'].get_output()` I get a theano variable instead.

Please make sure that the boxes below are checked before you submit your issue. Thank you!

- [x] Check that you are up-to-date with the master branch of Keras. You can update with:
pip install git+git://github.com/fchollet/keras.git --upgrade --no-deps

- [x] If running on Theano, check that you are up-to-date with the master branch of Theano. You can update with:
pip install git+git://github.com/Theano/Theano.git --upgrade --no-deps

- [ ] Provide a link to a GitHub Gist of a Python script that can reproduce your issue (or just copy the script here if it is short).

**Comment by fchollet** [comment]
This is covered in the Keras FAQ. You should define a Theano function with
the same input as the network input, and `m.nodes['emb'].get_output()` as
output.

**Comment by alyato** [comment]
@shyamupa  how do you solve it.

> the loss is not at all changing during training. I have tried changing learning rates, checking data sanity, but to no success

I meet the same question and don't know how to debug it ? Thanks.

**Comment by shyamupa** [comment]
I don't remember exactly but this usually happens when the network is somehow degenerated. Like one layer outputs always 1 and so derivatives go to zero.

**Comment by apinto92** [comment]
Hi, @fchollet 

I am kind of new to keras, so forgive me if I am sounding too rookie.

I want to train a CNN with 4 LSTMs in an end2end process. The goal is that after the CNN output I altered into 4 different orientations and then feed it to each of the LSTM. 

Is this possible to do when defining the model?

I tried something like this:

![dummy_ex](https://cloud.githubusercontent.com/assets/23172499/19860963/118d5b0c-9f8b-11e6-8da1-3e58ee800b54.png)

And after getting the output data transformation occurs and the LSTMs kick in

**Comment by monsieor** [comment]
@apinto92 did you find the answer?

**Comment by apinto92** [comment]
@monsieor  I was able to use a [function in Theano](http://deeplearning.net/software/theano/library/tensor/nnet/neighbours.html) that performs the partition of the image in subpatches. I developed my own Partition layer to be able to feed small sequences of the whole image, and incorporated in the model.
